### PR TITLE
refactor(plugin-text): move to @textlint

### LINKED ISSUE #455: Move core modules to @textlint namespace
npm recommened to use scoped.

> If you are prevented from publishing a package with a name too similar to an existing package, the easiest way to find a unique name is to use your scope. You can scope packages to your npm account by putting @ + your npm user name in front of the name. For example, my npm user name is ceejbot, so my scope is @ceejbot.
> http://blog.npmjs.org/post/168978377570/new-package-moniker-rules

We already have [@textlint](https://www.npmjs.com/org/textlint "@textlint") scope.

### Pros

- Make explicit

For example, [markdown-to-ast](https://github.com/textlint/textlint/tree/master/packages/markdown-to-ast "markdown-to-ast") is confused name.

### Cons

- Name chaning!

But, almost pacakges are only used in textlint.

- [textlint/packages/](https://github.com/textlint/textlint/tree/master/packages "textlint/packages/")

## Migration pacakges

| PR    | Before                     | After                                |
| :-----  | -------------------------- | ------------------------------------ |
| #473 | `txt-to-ast`               | `@textlint/text-to-ast`              |
| #460 | `markdown-to-ast`          | `@textlint/markdown-to-ast`          |
| #483 | `textlint-ast-test`        | `@textlint/ast-tester`               |
| #484 | `txt-ast-traverse`         | `@textlint/ast-traverse`             |
| #482 | `textlint-fixer-formatter` | `@textlint/fixer-formatter`          |
| #474 | `textlint-formatter`       | `@textlint/linter-formatter`         |
| #485 | `textlint-plugin-markdown` | `@textlint/textlint-plugin-markdown` |
| #486 | `textlint-plugin-text`     | `@textlint/textlint-plugin-text`     |

<details>
<summary> Migration Packages List </summary>

- [x] txt-to-ast -> `@textlint/text-to-ast` #473 
- [x] markdown-to-ast -> `@textlint/markdown-to-ast` #460 
- [x] textlint-ast-test -> `@textlint/ast-tester` #483
  - Add `er` is good
  - Align with [@textlint/ast-node-types](https://github.com/textlint/textlint/tree/master/packages/%40textlint/ast-node-types "@textlint/ast-node-types")
- [x] txt-ast-traverse -> `@textlint/ast-traverse` #484
  - Align with [@textlint/ast-node-types](https://github.com/textlint/textlint/tree/master/packages/%40textlint/ast-node-types "@textlint/ast-node-types")
- [x] textlint-fixer-formatter -> `@textlint/fixer-formatter` #482
- [x] textlint-formatter -> `@textlint/linter-formatter` #474
- [x] textlint-plugin-markdown -> `@textlint/textlint-plugin-markdown` #485
- [ ] textlint-plugin-text -> `@textlint/textlint-plugin-text` #486
  - `textlint-plugin` prefix is needed
  - [Package name convention](https://github.com/textlint/textlint/blob/master/docs/plugin.md#package-name-convention "Package name convention")

</details>

## No migration packages

These packages are used by external user.

- textlint
   - textlint it-self!
- gulp-textlint
   - textlint user use it
- textlint-tester
  - textlint rule creator use it

## ℹ️ Migration information for user

If you use someone of Migration pacakges, please move to scoped version.

For example, If you use `markdown-to-ast`, you migrate from `markdown-to-ast` to `@textlint/markdown-to-ast`. 

```
npm uninstall markdown-to-ast
npm install @textlint/markdown-to-ast
```

And, you should change `require("markdown-to-ast")` to `require("@textlint/markdown-to-ast")`.

-----

## How to migrate?

Migrate to `@textlint`.
<del>Maybe, [transfer-github-owner](https://github.com/azu/transfer-github-owner "transfer-github-owner") help this.</del>

Example: #473 #460 

1.  Move `<name>` to  `@textlint/<name>` directory 710a09e 
2. change usage of `<name>` to `@textlint/<name>` 5a12178 
  - And update `yarn.lock` by `yarn bootstrap`
3. Update core package table ec78494 
  - 💔 textlint-rule-no-dead-link found it as dead link
4. Add "publishConfig" 929b575 
  - `"publishConfig": { "access": "public" }` config is needed when you publish [Scoped packages](https://docs.npmjs.com/misc/scope "Scoped packages") at first time
  - Alt. `npm publish --access public` is anthor option
  - But, `publishConfig` is more reasonable.
5. `npm publish` it as `@textlint/<name>` from local
  - ❤️ This fix dead link of "Update core package table"
  - It create https://www.npmjs.com/package/@textlint/markdown-to-ast on npm.
6. Create PR

Finally, deprecate old pacakge:

```
npm deprecate <old-pacakge> "See https://github.com/textlint/textlint/issues/455"
```

### PR DESCRIPTION
**Pull request by 0x6b**
Finally closes #455!

Note: following line should be modified to `https://github.com/textlint/textlint/tree/master/packages/@textlint/textlint-plugin-text` after merge. I will open another pull request.

https://github.com/textlint/textlint/blob/29785dea1c91f3a6f818454a366a86033787346b/website/pages/en/index.js#L118

### CODE REVIEW

File: packages/textlint/test/textlint-module-resolver/textlint-module-resolver-test.js [hunk 1]
```diff
@@ -81,9 +81,9 @@ describe("textlint-module-resolver", function() {
     describe("#resolvePluginPackageName", function() {
         it("should resolve plugin package name", function() {
```
Comment (azu) [inline packages/textlint/test/textlint-module-resolver/textlint-module-resolver-test.js]: Originally, this test case aim to test non scoped plugin pacakge name.

Current test description should be renamed.

```diff
- should resolve plugin package name"
+ should resolve scoped plugin package name"
```

Comment (azu) [reply packages/textlint/test/textlint-module-resolver/textlint-module-resolver-test.js]: Can you add a test case that is non scoped test case using `textlint-plugin-html`.

```js
        it("should resolve plugin package name", function() {
            const resolver = createResolve();
            const shortPkg = resolver.resolvePluginPackageName("html");
            assert.equal(typeof shortPkg, "string");
            const longPkg = resolver.resolvePluginPackageName("textlint-plugin-html");
             assert.equal(typeof longPkg, "string");
             assert.equal(shortPkg, longPkg);
         });
```

Comment (0x6b) [reply packages/textlint/test/textlint-module-resolver/textlint-module-resolver-test.js]: @azu, thank you! Modify the description and add test case according to your review.

**Comment by 0x6b** [comment]
Alright, CI finished without error. Will deprecate the `textlint-plugin-text` and merge the PR.

**Comment by 0x6b** [comment]
```sh
npm deprecate textlint-plugin-text "See https://github.com/textlint/textlint/issues/455"
npm info --json textlint-plugin-text | jq .deprecated
"See https://github.com/textlint/textlint/issues/455"
```

**Pull request merged**